src/utils.ts

```diff
@@ -92,3 +92,26 @@ export function isBoolean(v: unknown): v is boolean {
 export function isNumber(v: unknown): v is number {
   return ((typeof v === "number" || (typeof v === "object" && v !== null && v.constructor === Number)) && !isNaN(v));
 }
+
+/**
+ * Generates a pipeline-friendly string
+ * that replaces non alphanumeric characters with dashes.
+ *
+ * @param applicationId The application if to sanitize.
+ *
+ * @returns The sanitized applicaiton id.
+ */
+export function sanitizeApplicationId(applicationId: string): string {
+  let result = "";
+  const tester = /([a-z0-9])/i;
+  for (let i = 0; i < applicationId.length; i++) {
+    if (tester.test(applicationId[i])) {
+      result += applicationId[i].toLowerCase();
+    } else {
+      if (result[result.length - 1] !== "-") {
+        result += "-";
+      }
+    }
+  }
```

Comment:
```suggestion
  return applicationId.replace(/[^A-Za-z0-9]/g, "-");
```

Wouldn't this work the same way?

Reply:
Not really because we skip when there is more than one alphanumeric in a row, so `bea@@@rizental` is transformed into `bea-rizental` instead of `bea---rizental`. Also you want `replaceAll` instead. This `return applicationId.replaceAll(/[a-z0-9]+/i, "-");` may work. I'll make the change :)

Reply:
`return applicationId.replace(/[a-z0-9]+/gi, "-");` worked :) `replaceAll` is not available for our target, but that is fine you were right, the `g` flag does the trick.